### PR TITLE
far2l-tty 2.8.0 (new formula)

### PR DESCRIPTION
**Pull request by Vladekk**
### Summary
  Add `far2l-tty`, the popular Linux/Unix port of the classic FAR Manager v2, TTY version.
                                                                                        
  ### Rationale
  far2l is widely used as the primary orthodox file manager for Linux/Unix              
  systems. While most distributions provide a heavy GUI build, this formula             
  focuses on a **lean, TTY-only version** suitable for servers and remote
  devices.                                                                              
                                                                  
  It is intentionally optimized for:                                                    
   
  - **Minimal footprint** — removes heavy dependencies like `aws-sdk-cpp`               
    (AWS/S3) and `samba` (SMB).                                   
  - **Environment versatility** — excludes X11 and wxWidgets, making it                 
    ideal for remote SSH sessions and headless servers.           
  - **Performance** — provides SFTP (via libssh), NFS, and WebDAV support               
    in a ~30 MB package.
                                                                                        
  ---                 
-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`?
- [x] Is your test running fine `brew test <formula>`?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- [x] AI was used to generate or assist with generating this PR. 
 
Claude (Anthropic) helped me to create the formula and iterated on it against `brew audit --new` and `brew audit --strict` until both passed cleanly.  I manually verified each change, ran `brew install --build-from-source`,  `brew test`, and `brew style` locally — all pass.

### CODE REVIEW

**Comment by Vladekk** [comment]
Existing far2l cask is a version of the same application rendered via GUI. My understanding that it was deprecated due to  Apple signing policies. Cask still can be revived if someone wants to do it, but this TTY version is useful regardless of the cask existence. I found that often you need far2l on a systems where installing from repositories is somehow complicated.

File: Formula/f/far2l-tty.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      -DCOLORER=ON
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", "-GNinja", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    assert_match "FAR2L Version:", shell_output("#{bin}/far2l --version")
```
Comment (botantony) [inline Formula/f/far2l-tty.rb]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/far2l --version")
```
Homebrew requires a simple test that checks if a program can do basic stuff correctly. From [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula):
> We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. However, a bad test is better than no test at all.

If it is not possible to test this program (f.e. it's a TUI), please add a comment explaining why there is only "version" test

Comment (Vladekk) [reply Formula/f/far2l-tty.rb]: This is indeed a TUI. I can try and add a test, but it will probably test some edge feature, not central to the app.

Comment (botantony) [reply Formula/f/far2l-tty.rb]: it's okay, just clarify in the comment why we cannot add a better test

**Comment by Vladekk** [comment]
This program is TUI, so in ita running state it is interactive, and mostly used this way.

While there CLI arguments do exist, none are well-suited to be used as a formula test. I'm open to implement tests later, if somebody more experienced will suggest what it might look like in this case.

Basic arguments are here:
https://manpages.ubuntu.com/manpages/noble/man1/far2l.1.html#backend-specific-options

File: Formula/f/far2l-tty.rb (hunk 1 above)
Comment (Vladekk) [reply Formula/f/far2l-tty.rb]: @botantony can you approve pls? I've added the comment. (Hopefully correctly)

Comment (botantony) [reply Formula/f/far2l-tty.rb]: I meant a code in code, something like
```suggestion
    # This is a TUI application
    assert_match "FAR2L Version:", shell_output("#{bin}/far2l --version")
```

Also let's assert the version as well:

```ruby
    assert_match version.to_s, shell_output("#{bin}/far2l --version")
```

Comment (Vladekk) [reply Formula/f/far2l-tty.rb]: @botantony done

**Comment by Vladekk** [comment]
@botantony sorry, missed what you said about version at first. I hope it is correct now.

File: Formula/f/far2l-tty.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      -DPYTHON=OFF
+      -DCOLORER=ON
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", "-GNinja", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+  # This is a TUI application, better tests are not possible
+  test do
```
Comment (botantony) [inline Formula/f/far2l-tty.rb]:
```suggestion
 
  test do
    # This is a TUI application, better tests are not possible
```

**Review by botantony — COMMENTED** [review]
LGTM. One more minor nitpick

**Review by botantony — APPROVED** [review]
Thank you!
